File: dragonsquad/craneam/com/Text.java
package dragon.craneam.com;

public class Text
  {
    String text;
    Buffers buffer;
    int x;
    int y;
    int fontId;

    public Text(int xPos, int yPos, int fontToUse)
      {
        buffer = new Buffers();
        x = xPos;
        y = yPos;
        fontId = fontToUse;
      }

    public void SetXPosition(int value)
      {
        x = value;
      }

    public void SetYPosition(int value)
      {
        y = value;
      }

    public Buffers GetBuffer()
      {
        return buffer;
      }

    public void SetText(String contents)
      {
        text = contents;
      }

    public void Render()
      {
        Font font;
        if (fontId == 1)
          {
            font = new Font(24);
          }
        else
          {
            font = new Font(15);
          }
        for (int i = 0; i < text.length(); i++)
          {
            switch (text.charAt(i))
              {
              case 'a':
                font.SetTextureLocation(0);
                break;
              case 'A':
                font.SetTextureLocation(0);
                break;
              case 'b':
                font.SetTextureLocation(1);
                break;
              case 'B':
                font.SetTextureLocation(1);
                break;
              case 'c':
                font.SetTextureLocation(2);
                break;
              case 'C':
                font.SetTextureLocation(2);
                break;
              case 'd':
                font.SetTextureLocation(3);
                break;
              case 'D':
                font.SetTextureLocation(3);
                break;
              case 'e':
                font.SetTextureLocation(4);
                break;
              case 'E':
                font.SetTextureLocation(4);
                break;
              case 'f':
                font.SetTextureLocation(5);
                break;
              case 'F':
                font.SetTextureLocation(5);
                break;
              case 'g':
                font.SetTextureLocation(6);
                break;
              case 'G':
                font.SetTextureLocation(6);
                break;
              case 'h':
                font.SetTextureLocation(7);
                break;
              case 'H':
                font.SetTextureLocation(7);
                break;
              case 'i':
                font.SetTextureLocation(8);
                break;
              case 'I':
                font.SetTextureLocation(8);
                break;
              case 'j':
                font.SetTextureLocation(9);
                break;
              case 'J':
                font.SetTextureLocation(9);
                break;
              case 'k':
                font.SetTextureLocation(10);
                break;
              case 'K':
                font.SetTextureLocation(10);
                break;
              case 'l':
                font.SetTextureLocation(11);
                break;
              case 'L':
                font.SetTextureLocation(11);
                break;
              case 'm':
                font.SetTextureLocation(12);
                break;
              case 'M':
                font.SetTextureLocation(12);
                break;
              case 'n':
                font.SetTextureLocation(13);
                break;
              case 'N':
                font.SetTextureLocation(13);
                break;
              case 'o':
                font.SetTextureLocation(14);
                break;
              case 'O':
                font.SetTextureLocation(14);
                break;
              case 'p':
                font.SetTextureLocation(15);
                break;
              case 'P':
                font.SetTextureLocation(15);
                break;
              case 'q':
                font.SetTextureLocation(16);
                break;
              case 'Q':
                font.SetTextureLocation(16);
                break;
              case 'r':
                font.SetTextureLocation(17);
                break;
              case 'R':
                font.SetTextureLocation(17);
                break;
              case 's':
                font.SetTextureLocation(18);
                break;
              case 'S':
                font.SetTextureLocation(18);
                break;
              case 't':
                font.SetTextureLocation(19);
                break;
              case 'T':
                font.SetTextureLocation(19);
                break;
              case 'u':
                font.SetTextureLocation(20);
                break;
              case 'U':
                font.SetTextureLocation(20);
                break;
              case 'v':
                font.SetTextureLocation(21);
                break;
              case 'V':
                font.SetTextureLocation(21);
                break;
              case 'w':
                font.SetTextureLocation(22);
                break;
              case 'W':
                font.SetTextureLocation(22);
                break;
              case 'x':
                font.SetTextureLocation(23);
                break;
              case 'X':
                font.SetTextureLocation(23);
                break;
              case 'y':
                font.SetTextureLocation(24);
                break;
              case 'Y':
                font.SetTextureLocation(24);
                break;
              case 'z':
                font.SetTextureLocation(25);
                break;
              case 'Z':
                font.SetTextureLocation(25);
                break;
              case '1':
                font.SetTextureLocation(26);
                break;
              case '2':
                font.SetTextureLocation(27);
                break;
              case '3':
                font.SetTextureLocation(28);
                break;
              case '4':
                font.SetTextureLocation(29);
                break;
              case '5':
                font.SetTextureLocation(30);
                break;
              case '6':
                font.SetTextureLocation(31);
                break;
              case '7':
                font.SetTextureLocation(32);
                break;
              case '8':
                font.SetTextureLocation(33);
                break;
              case '9':
                font.SetTextureLocation(34);
                break;
              case '0':
                font.SetTextureLocation(35);
                break;
              }
            if (i == 0)
              {
                font.Offset(x, y);
              }
            else
              {
                font.Offset(x - 34, 0);
              }
            font.AddIndices(buffer);
            font.AddVertexes(buffer);
            font.AddTextureCoordinates(buffer);
          }
        buffer.SetTextureId(3);
        buffer.CreateBuffers(true, true, true);
      }
  }

File: app/src/main/java/com/koodipuukko/dragonsquad/Text.kt
package com.koodipuukko.dragonsquad

class Text
	(xPos: Int, yPos: Int, fontToUse: Int) {
	private var text: String? = null
	private var buffer: Buffers = Buffers()
	var x: Int
	var y: Int
	private var fontId: Int

	init {
		x = xPos
		y = yPos
		fontId = fontToUse
	}

	fun setXPosition(value: Int) {
		x = value
	}

	fun setYPosition(value: Int) {
		y = value
	}

	fun getBuffer(): Buffers {
		return buffer
	}

	fun setText(contents: String?) {
		text = contents
	}

	fun render() {
		val font: Font = if(fontId == 1) {
			Font(24)
		} else {
			Font(15)
		}
		for(i in 0 until text!!.length) {
			when (text!![i]) {
				'a' -> font.SetTextureLocation(0)
				'A' -> font.SetTextureLocation(0)
				'b' -> font.SetTextureLocation(1)
				'B' -> font.SetTextureLocation(1)
				'c' -> font.SetTextureLocation(2)
				'C' -> font.SetTextureLocation(2)
				'd' -> font.SetTextureLocation(3)
				'D' -> font.SetTextureLocation(3)
				'e' -> font.SetTextureLocation(4)
				'E' -> font.SetTextureLocation(4)
				'f' -> font.SetTextureLocation(5)
				'F' -> font.SetTextureLocation(5)
				'g' -> font.SetTextureLocation(6)
				'G' -> font.SetTextureLocation(6)
				'h' -> font.SetTextureLocation(7)
				'H' -> font.SetTextureLocation(7)
				'i' -> font.SetTextureLocation(8)
				'I' -> font.SetTextureLocation(8)
				'j' -> font.SetTextureLocation(9)
				'J' -> font.SetTextureLocation(9)
				'k' -> font.SetTextureLocation(10)
				'K' -> font.SetTextureLocation(10)
				'l' -> font.SetTextureLocation(11)
				'L' -> font.SetTextureLocation(11)
				'm' -> font.SetTextureLocation(12)
				'M' -> font.SetTextureLocation(12)
				'n' -> font.SetTextureLocation(13)
				'N' -> font.SetTextureLocation(13)
				'o' -> font.SetTextureLocation(14)
				'O' -> font.SetTextureLocation(14)
				'p' -> font.SetTextureLocation(15)
				'P' -> font.SetTextureLocation(15)
				'q' -> font.SetTextureLocation(16)
				'Q' -> font.SetTextureLocation(16)
				'r' -> font.SetTextureLocation(17)
				'R' -> font.SetTextureLocation(17)
				's' -> font.SetTextureLocation(18)
				'S' -> font.SetTextureLocation(18)
				't' -> font.SetTextureLocation(19)
				'T' -> font.SetTextureLocation(19)
				'u' -> font.SetTextureLocation(20)
				'U' -> font.SetTextureLocation(20)
				'v' -> font.SetTextureLocation(21)
				'V' -> font.SetTextureLocation(21)
				'w' -> font.SetTextureLocation(22)
				'W' -> font.SetTextureLocation(22)
				'x' -> font.SetTextureLocation(23)
				'X' -> font.SetTextureLocation(23)
				'y' -> font.SetTextureLocation(24)
				'Y' -> font.SetTextureLocation(24)
				'z' -> font.SetTextureLocation(25)
				'Z' -> font.SetTextureLocation(25)
				'1' -> font.SetTextureLocation(26)
				'2' -> font.SetTextureLocation(27)
				'3' -> font.SetTextureLocation(28)
				'4' -> font.SetTextureLocation(29)
				'5' -> font.SetTextureLocation(30)
				'6' -> font.SetTextureLocation(31)
				'7' -> font.SetTextureLocation(32)
				'8' -> font.SetTextureLocation(33)
				'9' -> font.SetTextureLocation(34)
				'0' -> font.SetTextureLocation(35)
			}
			if(i == 0) {
				font.offset(x, y)
			} else {
				font.offset(x - 34, 0)
			}
			font.addIndices(buffer)
			font.addVertexes(buffer)
			font.addTextureCoordinates(buffer)
		}
		buffer.setTextureId(3)
		buffer.createBuffers(true, true, true)
	}
}
